### ISSUE TITLE
main_loop: Assertion `!always_readline' failed

### ISSUE THREAD

**Issue by irisjae** [hanslub42/rlwrap#179]
Hi, I've run into this error recently on my M1 machine, when wrapping my node script in rlwrap. Wrapping the same script with rlwrap works on Darwin but fails on Linux for some reason. 

```
rlwrap: main.c:477: main_loop: Assertion `!always_readline' failed.

rlwrap: Oops, crashed (caught SIGABRT) - this should not have happened!
If you need a core dump, re-configure with --enable-debug and rebuild
```

I'll investigate this problem in a bit. Have you got any pointers for me to look at? Thanks.

**Comment by hanslub42** [comment]
What is your exact `rlwrap` command line? And, while we're at it, what is the output of  `rlwrap -v` and `ldd $(which rlwrap)` ?

I see one possibility for this to happen (although I cannot provoke the crash that way) : if you specify  '-a' ('--always-readline') and '-N' ('--no-children'), and "waiting for a child" is the _first_ thing `rlwrap` encounters after startup

**Comment by irisjae** [comment]
@hanslub42 Hi, wondering if you've managed to get time to look into this?

In case you hadn't, perhaps if you have some suggestions of what kinds of changes would fix this issue (as I'd realised that I was no expert either on handling terminals), I'm happy to try my hands at this as well.

Thanks!

**Comment by hanslub42** [comment]
This is probably a duplicate of #191 (or rather, the other way round). Anyway, it should have been fixed by 4092e85ac8386ff9ae2bbf8580a1301446031a4a
